Assets/Mirror/Editor/SceneDrawer.cs

```diff
@@ -10,47 +11,31 @@ public override void OnGUI(Rect position, SerializedProperty property, GUIConten
         {
             if (property.propertyType == SerializedPropertyType.String)
             {
-                SceneAsset sceneObject = GetSceneObject(property.stringValue);
-                SceneAsset scene = (SceneAsset)EditorGUI.ObjectField(position, label, sceneObject, typeof(SceneAsset), true);
-                if (scene == null)
+                SceneAsset sceneObject = AssetDatabase.LoadAssetAtPath<SceneAsset>(property.stringValue);
+
+                if (sceneObject == null)
                 {
-                    property.stringValue = "";
+                    // try to load it from the build settings for legacy compatibility
+                    sceneObject = GetBuildSettingsSceneObject(property.stringValue);
                 }
-                else if (scene.name != property.stringValue)
+                if (sceneObject == null && !string.IsNullOrEmpty(property.stringValue))
                 {
-                    SceneAsset sceneObj = GetSceneObject(scene.name);
-                    if (sceneObj == null)
-                    {
-                        Debug.LogWarning("The scene " + scene.name + " cannot be used. To use this scene add it to the build settings for the project");
-                    }
-                    else
-                    {
-                        property.stringValue = scene.name;
-                    }
+                    Debug.LogError($"Could not find scene {property.stringValue} in {property.propertyPath}, assign the proper scenes in your NetworkManager");
                 }
+                SceneAsset scene = (SceneAsset)EditorGUI.ObjectField(position, label, sceneObject, typeof(SceneAsset), true);
+
+                property.stringValue = AssetDatabase.GetAssetPath(scene);
             }
             else
             {
                 EditorGUI.LabelField(position, label.text, "Use [Scene] with strings.");
             }
         }
 
-        protected SceneAsset GetSceneObject(string sceneObjectName)
+        protected SceneAsset GetBuildSettingsSceneObject(string sceneName)
         {
-            if (string.IsNullOrEmpty(sceneObjectName))
-            {
-                return null;
-            }
-
-            foreach (EditorBuildSettingsScene editorScene in EditorBuildSettings.scenes)
-            {
-                if (editorScene.path.IndexOf(sceneObjectName) != -1)
-                {
-                    return AssetDatabase.LoadAssetAtPath(editorScene.path, typeof(SceneAsset)) as SceneAsset;
-                }
-            }
-            Debug.LogWarning("Scene [" + sceneObjectName + "] cannot be used. Add this scene to the 'Scenes in the Build' in build settings.");
-            return null;
+            Scene scene = SceneManager.GetSceneByName(sceneName);
```

Comment:
`GetSceneByName` only works on loaded scenes, need to use `EditorBuildSettingsScene `
```csharp
            foreach (EditorBuildSettingsScene buildScene in EditorBuildSettings.scenes)
            {
                SceneAsset sceneAsset = AssetDatabase.LoadAssetAtPath<SceneAsset>(buildScene.path);
                if (sceneAsset.name == sceneName)
                {
                    return sceneAsset;
                }
            }
```